extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskQueryMaker.java

```diff
@@ -227,7 +240,9 @@ public QueryResponse<Object[]> runQuery(final DruidQuery druidQuery)
       );
     } else {
       if (ctxDestination != null && !DESTINATION_REPORT.equals(ctxDestination)) {
-        throw new IAE("Cannot SELECT with destination [%s]", ctxDestination);
+        throw DruidException.forPersona(DruidException.Persona.DEVELOPER)
+                            .ofCategory(DruidException.Category.DEFENSIVE)
+                            .build("Cannot SELECT with destination [%s]", ctxDestination);
```

Comment:
Regarding this, I saw the SYSTEM_CONTEXT has the ctxDestination key, therefore I assumed that it is set by Druid, hence reaching here would have been incorrect. There were some changes merged later which I'll go through to see if this is set internally or by the user.

Reply:
Rechecked this code, the "destination" query context is no longer valid, therefore removing these checks and the exception (cc: @cryptoe since its related to your changes)